ckanext/external_storage/validators.py

```diff
@@ -0,0 +1,25 @@
+from ckan.plugins.toolkit import Invalid
+
+
+def upload_has_sha256(key, flattened_data, errors, context):
+    if flattened_data[key] == 'upload':
+        __extras_key = (key[0], key[1], '__extras')
+        if __extras_key not in flattened_data:
+            raise Invalid("Resource's sha256 field cannot be missing for uploads.")
+        elif 'sha256' not in flattened_data[__extras_key]:
```

Comment:
Would be nice to validate that this is an actual sha256 - see https://github.com/datopian/metastore-lib/blob/master/metastore/util.py#L28 for some really simple implementation of something that checks a string is sha256 like, I suppose this can be copied (and simplified to be sha256 specific).

Reply:
Done!